### PR TITLE
Added tests for collision avoidance using CHOMP planner

### PR DESCRIPTION
**Pull request by raghavendersahdev**
This pull requests has a few collision testing scripts and the obstacle_cost_weight parameter is reduced.

### CODE REVIEW

**Comment by davetcoleman** [comment]
Also @raghavendersahdev I would reword the title of your PR in the future to say "demos" instead of "tests" since there is no formal test coverage / integration tests / unit tests

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ You should also have gone through the steps in `Visualization with MoveIt! RViz
 
 Prerequisites
 --------------
- 1. You must have the latest version of MoveIt! installed. On ROS Kinetic you will need to build MoveIt! from source. We will go through the steps for doing this below.
+ 1. You must have the latest version of MoveIt! installed. On ROS Kinetic you will need to build MoveIt! from source. A build from source is required as CHOMP is not officially released so ``apt-get install`` for moveIt would not be appropriate here. We will go through the steps for doing this below.
```
Comment (simonschmeisser) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: MoveIt! instead of moveIt,  "as CHOMP is not part of the official release yet. It is therefore not included in the binary packages."

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 2]
```diff
@@ -49,10 +49,24 @@ Using CHOMP with Your Robot
       <arg name="debug" value="$(arg debug)"/>
       <arg name="planner" value="chomp" />
     </include>
-#. Open the ``move_group.launch`` file in your ``<robot_moveit_config>/launch/`` folder and make two changes. First, add ``<arg name="planner" default="ompl" />`` just under the ``<launch>`` tag and second, within the ``<include ns="move_group">`` tag replace ``<arg name="pipeline" value="ompl" />`` with ``<arg name="pipeline" value="$(arg planner)" />``.
+#. Open the ``move_group.launch`` file in your ``<robot_moveit_config>/launch/`` folder and make two changes.
+
+ 7.1. First, add ``<arg name="planner" default="ompl" />`` just under the ``<launch>`` tag, and,
```
[the remark this answers is not in the document]
Comment (davetcoleman) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: I requested they be subindices because previously there were multiple steps in the same sentence. I thought it would be nice since they both are edits to one file, but I don't feel strongly about this.

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 3]
```diff
@@ -70,3 +70,57 @@ To run the CHOMP planner with obstacles, do the following in two seperate termin
 
   roslaunch panda_moveit_config demo_chomp.launch
   python collision_scene_test_1.py OR python collision_scene_test_2.py
+
+
+Tweaking some of the parameters for CHOMP
+-----------------------------------------
+Chomp has some optimization parameters associated with it. These can be modified for the given environment/robot you are working with and is normally present in the `chomp_planning.yaml <https://github.com/ros-planning/panda_moveit_config/blob/master/config/chomp_planning.yaml>`_ file in config folder of the robot you are working with. If this file does not exist for your robot, you can create it and set the parameter values as you want. Following are some of the insights to set up these parameter values for some of them:
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: Chomp -> "CHOMP"

"The following are some..."

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 4]
```diff
@@ -49,10 +49,78 @@ Using CHOMP with Your Robot
...
 ----------------
 If you have the ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository you should be able to simply run the demo: ::
 
-  roslaunch panda_moveit_config demo_chomp.launch
\ No newline at end of file
+  roslaunch panda_moveit_config demo_chomp.launch
+
+Testing CHOMP with Obstacles in the Scene
+-----------------------------------------
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts (`collision_scene_test_1.py <./collision_scene_test1.py>`_ or `collision_scene_test_2.py <./collision_scene_test2.py>`_). The first scripts creates a complex scene with four ostacles. The second script creates a simple environment with one obstacle. One can change the position/size of the obstacles to change the scene.
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: can you move the code snippets into their own line, instead of in-line with paragraph?

File: doc/chomp_interface/collision_scene_test_2.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import rospy
+from moveit_commander import RobotCommander, PlanningSceneInterface
+import geometry_msgs.msg
+import time
+
+
+class CreateScene1(object):
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/collision_scene_test_2.py]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 3 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 4 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/collision_scene_test_1.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import rospy
+from moveit_commander import RobotCommander, PlanningSceneInterface
+import geometry_msgs.msg
+import time
+
+
+class CreateScene1(object):
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/collision_scene_test_1.py]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 1 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (davetcoleman) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: @raghavendersahdev please see "mixed list" https://en.support.wordpress.com/markdown-quick-reference/

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 7]
```diff
@@ -49,10 +49,24 @@ Using CHOMP with Your Robot
...
+
+Testing CHOMP with Obstacles in the Scene
+-----------------------------------------
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts (`collision_scene_test_1.py <./collision_scene_test1.py>`_ or `collision_scene_test_2.py <./collision_scene_test2.py>`_). The first scripts creates a complex scene with four ostacles. The second script creates a simple environment with one obstacle. One can change the position/size of the obstacles to change the scene. 
+
+
+To run the CHOMP planner with obstacles, do the following in two seperate terminals: ::
+
+  roslaunch panda_moveit_config demo_chomp.launch
+  python collision_scene_test_1.py OR python collision_scene_test_2.py
```
[the remark this answers is not in the document]
Comment (davetcoleman) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: @raghavendersahdev

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 8]
```diff
@@ -49,10 +49,81 @@ Using CHOMP with Your Robot
...
+  roslaunch panda_moveit_config demo_chomp.launch
+
+Testing CHOMP with Obstacles in the Scene
+-----------------------------------------
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts 
+ 
+ - `collision_scene_test_1.py <./scripts/collision_scene_test_1.py>`_ or 
+ - `collision_scene_test_2.py <./scripts/collision_scene_test_2.py>`_).
+
+ The first scripts creates a complex scene with four ostacles. The second script creates a simple environment with one obstacle. One can change the position/size of the obstacles to change the scene.
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: remove space at start of sentence

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 9]
```diff
@@ -49,10 +49,81 @@ Using CHOMP with Your Robot
...
 ----------------
 If you have the ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository you should be able to simply run the demo: ::
 
-  roslaunch panda_moveit_config demo_chomp.launch
\ No newline at end of file
+  roslaunch panda_moveit_config demo_chomp.launch
+
+Testing CHOMP with Obstacles in the Scene
+-----------------------------------------
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: end sentence with ``: ::``

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 10]
```diff
@@ -49,10 +49,81 @@ Using CHOMP with Your Robot
...
+ The first scripts creates a complex scene with four ostacles. The second script creates a simple environment with one obstacle. One can change the position/size of the obstacles to change the scene. 
+
+
+To run the CHOMP planner with obstacles, do the following in two seperate terminals: ::
+
+  roslaunch panda_moveit_config demo_chomp.launch
+  python collision_scene_test_1.py OR python collision_scene_test_2.py
+
+
+Tweaking some of the parameters for CHOMP
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: Capitalize consistently

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 11]
```diff
@@ -49,10 +49,81 @@ Using CHOMP with Your Robot
...
+
+Choosing Parameters for CHOMP requires some sort of intuition based on the environment we are working in. One can have the default parameters for CHOMP and this works well in environments without  obstacles. However in cases where the scene is populated with obstacles, we need to vary some parameters to ensure that CHOMP is not stuck in local minima, or quickly finds optimal solutions, prefering trajectories which ovoids obstacles. Some parameters like increasing the *ridge_factor* to say 0.001 makes CHOMP avoids obstacles by not prefering smooth trajectories, so there is a trade-off  between smoothness and CHOMP's ability to avoid obstacles. Choosing the correct number of *max_iterations*, *learning_rate* is important based on the environment we are working in. Not choosing the appropriate CHOMP parameters might lead to CHOMP reporting not finding a collision free path. *collision_clearance*, *collision_threshold* parameters are useful in specifying the minimum distance to be kept from obstacles to avoid collisions.
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: "Parameters" -> "parameters"

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 12]
```diff
@@ -49,10 +49,81 @@ Using CHOMP with Your Robot
...
+
+OMPL is a open source library for sampling based / randomized motion planning algorithms. Sampling based algorithms are probabilistically complete: a solution would be eventually found if one exists, however non-existence of a solution cannot be reported. These algorithms are efficient and usually find a solution quickly. OMPL does not contain any code related to collision checking or visualization as the designers of OMPL did not want to tie it to a any particular colision checker or visualization front end. The library is designed so it can be easily integrated into systems that provide the additional components. MoveIt integrates directly with  OMPL and uses the motion planners from OMP as its default set of planners. The planners in OMPL are abstract; i.e. OMPL has no concept of a robot. Instead, MoveIt! configures OMPL and provides the back-end for OMPL to work with problems in Robotics.
+
+
```
Comment (davetcoleman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: remove double line break

File: doc/chomp_interface/collision_scene_test_1.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (davetcoleman) [reply doc/chomp_interface/collision_scene_test_1.py]: did you push the changes?

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 11 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 12 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/collision_scene_test_1.py [hunk 13]
```diff
@@ -0,0 +1,60 @@
+import rospy
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/collision_scene_test_1.py]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 8 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: removed the space at the start of the sentence. "The first scripts...."

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 9 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: if I end this sentence with : :: the formatting for the bullets below this line gets messed up like: 
selection_024

Currently I just end it with :

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 14]
```diff
@@ -1,10 +1,10 @@
-CHOMP Interface
+CHOMP Planner
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: Thanks @mlautman  , I just rebased the PR to reflect the latest commits. Is that fine?

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 15]
```diff
@@ -49,10 +49,77 @@ Using CHOMP with Your Robot
...
+-----------------------------------------
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts:
+
+- `collision_scene_test_1.py <./scripts/collision_scene_test_1.py>`_ or
+
+- `collision_scene_test_2.py <./scripts/collision_scene_test_2.py>`_.
+
+The first scripts creates a complex scene with four ostacles. The second script creates a simple environment with one obstacle. One can change the position/size of the obstacles to change the scene. 
+
+To run the CHOMP planner with obstacles, do the following in two seperate terminals: ::
```
Comment (mlautman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: For consistency with the other tutorials can you reword to:

Open two shells. In the first shell start RViz and wait for everything to finish loading: ::

  roslaunch panda_moveit_config demo_chomp.launch

In the second shell, run either of the two commands: ::

  rosrun moveit_tutorials collision_scene_test_1.py

or: ::

  rosrun moveit_tutorials collision_scene_test_2.py

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 16]
```diff
@@ -49,10 +49,77 @@ Using CHOMP with Your Robot
...
+
+  roslaunch panda_moveit_config demo_chomp.launch
+  rosrun moveit_tutorials collision_scene_test_1.py OR rosrun moveit_tutorials collision_scene_test_2.py
```
Comment (mlautman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: Can you rename these something more descriptive. EG: chomp_collision_scene_example_cluttered.py and chomp_collision_scene_example_sparse.py

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 17]
```diff
@@ -49,10 +49,77 @@ Using CHOMP with Your Robot
...
+-----------------------------------------
+CHOMP has some optimization parameters associated with it. These can be modified for the given environment/robot you are working with and is normally present in the `chomp_planning.yaml <https://github.com/ros-planning/panda_moveit_config/blob/master/config/chomp_planning.yaml>`_ file in config folder of the robot you are working with. If this file does not exist for your robot, you can create it and set the parameter values as you want. The following are some of the insights to set up these parameter values for some of them:
+
+- *planning_time_limit*: the maximum time the optimizer can take to find a solution before terminating
+
+- *max_iterations*: this is the maximum number of iterations that the planner can take to find a good solution while optimization
+
+- *max_iterations_after_collision_free*: maximum iterations to be performed after a collision free path is found.
+
+- *smoothness_cost_weight*:  the smoothness_cost_weight parameters controls its weight in the final cost that CHOMP is actually optimizing over
```
Comment (mlautman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: one space

File: doc/chomp_interface/scripts/collision_scene_test_1.py [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+        p.pose.position.x = pose[0]
+        p.pose.position.y = pose[1]
+        p.pose.position.z = pose[2]
+        p.pose.orientation.x = pose[3]
+        p.pose.orientation.y = pose[4]
+        p.pose.orientation.z = pose[5]
+        p.pose.orientation.w = pose[6]
+
+        self._scene.add_box(name, p, (dimensions[0], dimensions[1], dimensions[2]))
+	print "============ Waiting while RVIZ displays the scene with four obstacles..."
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_test_1.py]: formatting

File: doc/chomp_interface/scripts/collision_scene_test_1.py [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+        rospy.sleep(1)
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_test_1.py]: Please don't use sleep. See the move_group_python_tutorial for explicitly waiting for the ps to update

File: doc/chomp_interface/scripts/collision_scene_test_2.py [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+        p.header.stamp = rospy.Time.now()
+        p.pose.position.x = pose[0]
+        p.pose.position.y = pose[1]
+        p.pose.position.z = pose[2]
+        p.pose.orientation.x = pose[3]
+        p.pose.orientation.y = pose[4]
+        p.pose.orientation.z = pose[5]
+        p.pose.orientation.w = pose[6]
+        self._scene.add_box(name, p, (dimensions[0], dimensions[1], dimensions[2]))
+	print "============ Waiting while RVIZ displays the scene with one obstacle..."
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_test_2.py]: same as above

File: doc/chomp_interface/scripts/collision_scene_test_2.py [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+        rospy.sleep(1)
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_test_2.py]: same as above

File: doc/chomp_interface/scripts/collision_scene_test_2.py [hunk 22]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+import rospy
+from moveit_commander import RobotCommander, PlanningSceneInterface
+import geometry_msgs.msg
+import time
+
+class CreateScene(object):
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_test_2.py]: better class name. Why is this different from the above file?

File: doc/chomp_interface/scripts/collision_scene_test_1.py [hunk 23]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+import rospy
+from moveit_commander import RobotCommander, PlanningSceneInterface
+import geometry_msgs.msg
+import time
+
+
+class CreateScene(object):
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_test_1.py]: better class name.

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 15 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done and pushed the changes

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 24]
```diff
@@ -49,10 +49,77 @@ Using CHOMP with Your Robot
...
 Running the Demo
 ----------------
 If you have the ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository you should be able to simply run the demo: ::
 
-  roslaunch panda_moveit_config demo_chomp.launch
\ No newline at end of file
+  roslaunch panda_moveit_config demo_chomp.launch
+
+Testing CHOMP with Obstacles in the Scene
+-----------------------------------------
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done and pushed the changes

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 25]
```diff
@@ -49,10 +49,77 @@ Using CHOMP with Your Robot
...
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts:
+
+- `collision_scene_test_1.py <./scripts/collision_scene_test_1.py>`_ or
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: fixed

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 16 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done and pushed changes

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 17 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: fixed

File: doc/chomp_interface/scripts/collision_scene_test_1.py (hunk 19 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_test_1.py]: I removed this particular sleep usage, but I still got the sleep usage over [here](https://github.com/raghavendersahdev/moveit_tutorials/blob/kinetic-devel/doc/chomp_interface/scripts/collision_scene_example_sparse.py#L17). to show the user some sort of terminal output while loading the scene.The [move_group_python_tutorial](http://docs.ros.org/indigo/api/moveit_tutorials/html/doc/pr2_tutorials/planning/scripts/doc/move_group_python_interface_tutorial.html) for pr2 also used sleep like : 
selection_025
selection_026

File: doc/chomp_interface/scripts/collision_scene_test_2.py (hunk 20 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_test_2.py]: done

File: doc/chomp_interface/scripts/collision_scene_test_2.py (hunk 22 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_test_2.py]: changed class name to ``CreateSparseScene`` should I keep the filename same as the class name or is this fine as is,  filename = ``collision_scene_example_sparse.py``?

File: doc/chomp_interface/scripts/collision_scene_test_1.py (hunk 23 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_test_1.py]: changed class name to ``CreateClutteredScene`` should I keep the filename same as the class name or is this fine as is, filename = ``collision_scene_example_cluttered.py``?

File: doc/chomp_interface/collision_scene_test_1.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/collision_scene_test_1.py]: yes

File: doc/chomp_interface/scripts/collision_scene_test_2.py (hunk 21 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_test_2.py]: replied for sleep as above.

File: doc/chomp_interface/scripts/collision_scene_test_2.py (hunk 22 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_test_2.py]: currently have class-name same as file-name

**Comment by davetcoleman** [comment]
See Travis build errors:

```

- ./build/html/doc/chomp_interface/chomp_interface_tutorial.html
  *  internally linking to ./scripts/collision_scene_example_cluttered.py, which does not exist (line 601)
     <a class="reference external" href="./scripts/collision_scene_example_cluttered.py">collision_scene_example_cluttered.py</a>
  *  internally linking to ./scripts/collision_scene_example_sparse.py, which does not exist (line 602)
     <a class="reference external" href="./scripts/collision_scene_example_sparse.py">collision_scene_example_sparse.py</a>
htmlproofer 3.9.1 | Error:  HTML-Proofer found 2 failures!
```

File: doc/chomp_interface/scripts/collision_scene_example_cluttered.py [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+        # clear the scene
+        self._scene.remove_world_object()
+
+        self.robot = RobotCommander()
+
+        # pause to wait for rviz to load        
+        print "============ Waiting while RVIZ displays the scene with four obstacles..."
+        rospy.sleep(4)
+
+        floor_pose = [0, 0, -1.12, 0, 0, 0, 1]
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_example_cluttered.py]: From here on should be a separate method not part of the init.

File: doc/chomp_interface/scripts/collision_scene_example_sparse.py [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+    def __init__(self):
+        self._scene = PlanningSceneInterface()
+
+        # clear the scene
+        self._scene.remove_world_object()
+        self.robot = RobotCommander()
+
+        # pause to wait for rviz to load
+        print "============ Waiting while RVIZ displays the scene with one obstacle..."
+        rospy.sleep(4)
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_example_sparse.py]: same as above

File: doc/chomp_interface/scripts/collision_scene_example_sparse.py [hunk 28]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+import rospy
+from moveit_commander import RobotCommander, PlanningSceneInterface
+import geometry_msgs.msg
+import time
+
+class CollisionSceneExampleSparse(object):
```
Comment (mlautman) [inline doc/chomp_interface/scripts/collision_scene_example_sparse.py]: This is a ton of code duplication. You could achieve this in a single script that takes in a command line argument telling it which scene to load

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 29]
```diff
@@ -49,10 +49,82 @@ Using CHOMP with Your Robot
...
+  roslaunch panda_moveit_config demo_chomp.launch
+
+In the second shell, run either of the two commands: ::
+
+  rosrun moveit_tutorials collision_scene_example_cluttered.py
+
+or: ::
+
+  rosrun moveit_tutorials collision_scene_example_sparse.py
+
```
Comment (mlautman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: Add instructions telling the user to select CHOMP in the MotionPlanning pannel under the Context tab.
You will also want to tell them to move the endeffector around with the imarker and hit plan

File: doc/chomp_interface/scripts/collision_scene_example_cluttered.py [hunk 30]
```diff
@@ -0,0 +1,62 @@
...
+        self._scene = PlanningSceneInterface()
+
+        # clear the scene
+        self._scene.remove_world_object()
+
+        self.robot = RobotCommander()
+
+        # pause to wait for rviz to load        
+        print "============ Waiting while RVIZ displays the scene with four obstacles..."
+        rospy.sleep(4)
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_example_cluttered.py]: Hi Mike, if I don't put any sleep, the obstacles do not show up in RViz, hence I used a sleep over here, I could reduce it to rospy.sleep(2), any reasons for not using sleep ?

[the remark this answers is not in the document]
Comment (mlautman) [reply doc/chomp_interface/scripts/collision_scene_example_cluttered.py]: Sleep in this context is a hack that masks the bigger issue. Instead of blindly sleeping, this demo script should wait for the correct topics to be available before publishing to the planning scene interface and it should wait for those changes to be reflected in the planning scene before returning.

[the remark this answers is not in the document]
Comment (mlautman) [reply doc/chomp_interface/scripts/collision_scene_example_cluttered.py]: Waiting an arbitrary number of seconds is not an approach that we want to encourage among the users of MoveIt!

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 31]
```diff
@@ -49,10 +49,82 @@ Using CHOMP with Your Robot
...
 
 Running the Demo
 ----------------
 If you have the ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository you should be able to simply run the demo: ::
 
-  roslaunch panda_moveit_config demo_chomp.launch
\ No newline at end of file
+  roslaunch panda_moveit_config demo_chomp.launch
+
+Testing CHOMP with Obstacles in the Scene
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 32]
```diff
@@ -49,10 +49,82 @@ Using CHOMP with Your Robot
...
++++++++++++++++++++++++++++++++++++++++++
+To test CHOMP in an evironment with obstacles, you can run any of the sample python scripts:
```
[the remark this answers is not in the document]
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done

File: doc/chomp_interface/scripts/collision_scene_example_cluttered.py (hunk 26 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_example_cluttered.py]: done

File: doc/chomp_interface/scripts/collision_scene_example_sparse.py (hunk 28 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/scripts/collision_scene_example_sparse.py]: removed duplicate code and merged into single file and behavior now depends on command line argument ("cluttered" or "sparse")

File: doc/chomp_interface/chomp_interface_tutorial.rst [hunk 33]
```diff
@@ -1,58 +0,0 @@
-CHOMP Interface
```
Comment (mlautman) [inline doc/chomp_interface/chomp_interface_tutorial.rst]: Note: If you use `git mv` rather than `mv` when renaming these files then git does a better job of tracking your changes vs the original.

File: doc/chomp_planner/scripts/collision_scene_example.py [hunk 34]
```diff
@@ -0,0 +1,82 @@
...
+    rospy.init_node("collision_scene_example_cluttered")
+    while not rospy.search_param('robot_description_semantic') and not rospy.is_shutdown():
+        time.sleep(0.5)
+    load_scene = CollisionSceneExample()
+
+    if (len(sys.argv) != 2):
+        print "Correct usage:: \n\"rosrun moveit_tutorials collision_scene_example.py cluttered\" OR \n\"rosrun moveit_tutorials collision_scene_example.py sparse\""
+        sys.exit()
+    if sys.argv[1] == "cluttered": 
+        load_scene.add_four_boxes();
```
Comment (mlautman) [inline doc/chomp_planner/scripts/collision_scene_example.py]: Nice!

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 33 above)
Comment (mlautman) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: Also, you need to test with the `build_locally.sh` script before each push. You didn't rename the tutorial in `index.rst` so the chomp tutorial isn't getting built

File: doc/chomp_planner/scripts/collision_scene_example.py [hunk 35]
```diff
@@ -0,0 +1,82 @@
...
+    elif sys.argv[1] == "sparse":
+        load_scene.add_one_box();
+    else:
+        print "Please specify correct type of scene as cluttered or sparse"
+        sys.exit()
+    rospy.spin()
```
Comment (mlautman) [inline doc/chomp_planner/scripts/collision_scene_example.py]: This node does not need to spin. It should die after it has loaded the scene. Please see my earlier comments about waiting for updates in the planning scene.

File: doc/chomp_interface/chomp_interface_tutorial.rst (hunk 33 above)
Comment (raghavendersahdev) [reply doc/chomp_interface/chomp_interface_tutorial.rst]: done. I tested the `build_locally.sh` script and renamed the CHOMP tutorial in the `index.rst` file.

File: doc/chomp_planner/scripts/collision_scene_example.py (hunk 35 above)
Comment (raghavendersahdev) [reply doc/chomp_planner/scripts/collision_scene_example.py]: thanks @mlautman  yes this makes sense, no need to call spin. I removed spin call here.

**Comment by mlautman** [comment]
see travis build failure

**Comment by davetcoleman** [comment]
Yes this is a known limitation of our current travis setup - if you add new files / html pages it complains that page is not available on the live website. Would be nice if someone could address.

@mlautman I think we should merge this

File: index.rst [hunk 36]
```diff
@@ -66,7 +66,7 @@ Configuration
    doc/kinematics_configuration/kinematics_configuration_tutorial
    doc/custom_constraint_samplers/custom_constraint_samplers_tutorial
    doc/ompl_interface/ompl_interface_tutorial
-   doc/chomp_interface/chomp_interface_tutorial
+   doc/chomp_planner/chomp_planner_tutorial
```
Comment (davetcoleman) [inline index.rst]: @mlautman lets no do any more changes in the PR, this one is fine, but in the future I think you should *avoid* asking people to rename folders and filenames in these tutorials, as that breaks hyperlinks on a live website which should be generally avoided

Comment (mlautman) [reply index.rst]: I agree about merging. With regard to filenames and directories, I think that it's important that we maintain the standards in the contributing guidelines. That said, this tutorial could have kept the name `chomp_interface_tutorial` and I wouldn't have said anything. In the future, I'll probably just suggest that the name of the tutorial stays the same rather than suggest that the file and dir name are also changed to match.
